### PR TITLE
Fixes @a[r=X] & Allows the plugin to work with 1.7-1.17

### PR DESCRIPTION
**Pull request by GrangerTheDog**
Fixes the error that happens when the Location and Player are not in the same world in the isInRange function!

Reported by 4P5#1749 in the Discord channel!
Caused by: java.lang.IllegalArgumentException: Cannot measure distance between world and world_nether

### CODE REVIEW

**Comment by GrangerTheDog** [comment]
Added some more stuff! Like the ability to use with 1.7-1.17

**Review by TehBrian — CHANGES_REQUESTED** [review]

File: plugin/pom.xml [hunk 1]
```diff
@@ -83,22 +96,35 @@
...
+                                <includes>
+                                    <include>com.oliverdunk</include>
+                                    <include>io.socket</include>
+                                    <include>com.squareup.*</include>
+                                    <include>net.kyori</include>
+                                    <include>org.json</include>
+                                    <include>org.jetbrains*</include>
+
+
+
```
Comment (TehBrian) [inline plugin/pom.xml:133]: unnecessary newlines

File: plugin/src/main/java/net/mcjukebox/plugin/bukkit/MCJukebox.java [hunk 2]
```diff
@@ -29,11 +31,21 @@
     @Getter private ShowManager showManager;
     @Getter private TimeUtils timeUtils;
 
+
+    private BukkitAudiences adventure;
+
+    public @NonNull BukkitAudiences adventure() {
+        if(this.adventure == null) {
+            throw new IllegalStateException("Tried to access Adventure when the plugin was disabled!");
```
Comment (TehBrian) [inline plugin/src/main/java/net/mcjukebox/plugin/bukkit/MCJukebox.java:39]: nitpick, but the if statement style should match the rest of the project (space after `if`)

File: plugin/src/main/java/net/mcjukebox/plugin/bukkit/utils/SpigotUtils.java [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
 		String URL = langManager.get("user.openDomain") + "?token=" + token;
-
-		TextComponent message = new TextComponent(TextComponent.fromLegacyText(langManager.get("user.openClient")));
-		message.setClickEvent(new ClickEvent(ClickEvent.Action.OPEN_URL, URL));
-		message.setHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, TextComponent.fromLegacyText(langManager.get("user.openHover"))));
-		player.spigot().sendMessage(message);
+		TextComponent message = Component.text()
+				.append(LegacyComponentSerializer.legacy('&').deserialize(langManager.get("user.openClient")))
+				.clickEvent(ClickEvent.openUrl(URL))
+				.hoverEvent(HoverEvent.showText(LegacyComponentSerializer.legacy('&').deserialize(langManager.get("user.openClient")))).build();
```
Comment (TehBrian) [inline plugin/src/main/java/net/mcjukebox/plugin/bukkit/utils/SpigotUtils.java:20]: `.legacy('&')` can be replaced with `.legacyAmpersand()`

File: shared/pom.xml [hunk 4]
```diff
@@ -33,4 +33,5 @@
         </dependency>
     </dependencies>
 
+
```
Comment (TehBrian) [inline shared/pom.xml:53]: unnecessary newline addition

**Comment by oliverdunk** [comment]
Sorry for taking far too long with this one - I've extracted the bug fix into https://github.com/oliverdunk/JukeboxAPI/commit/b528fe305b084e4a31bf6026abc8924a1cbf0821. I'm going to close this since I think the rest of the changes are related to the volume work you've done which isn't currently supported by the public client.

**Pull request closed without merging**